pwiz_tools/Skyline/Controls/Graphs/GraphChromatogram.cs

```diff
@@ -2282,11 +2283,29 @@ private class DisplayPeptide
                     if (listTimes.Count > 0)
                         chromGraphPrimary.RetentionMsMs = listTimes.ToArray();
 
-                    if (settings.PeptideSettings.Libraries.HasMidasLibrary)
+                    if (settings.PeptideSettings.Libraries.HasMidasLibrary && _stateProvider.SelectedNode != null)
                     {
-                        chromGraphPrimary.MidasRetentionMsMs = settings.PeptideSettings.Libraries.MidasLibraries.SelectMany(
-                            lib => lib.GetSpectraByPrecursor(chromGraphPrimary.Chromatogram.FilePath, chromGraphPrimary.Chromatogram.PrecursorMz))
-                            .Select(s => s.RetentionTime).ToArray();
+                        PeptideDocNode nodePep = null;
+                        TransitionGroupDocNode nodeTranGroup = null;
+                        switch (_stateProvider.SelectedNode)

```

Comment:
You could also have done:
nodePep = _stateProvider.GetNodeOfType<PeptideTreeNode>()?.DocNode
nodeTranGroup = _stateProvider.GetNodeOfType<TransitionGroupTreeNode>()?.DocNode